lib/stellar/sep10.rb

```diff
@@ -262,6 +262,8 @@ def self.verify_challenge_tx_threshold(
       signers.each do |s|
         if !signer_strs_found.include?(s['key'])
           next
+        else
+          signer_strs_found.delete(s['key'])
         end
         signers_found.add(s)
         weight += s['weight']
```

Comment:
What's the goal of having the delete inside the `else`? Everything else that runs in this block is essentially the `else` case because inside the if we call `next` that skips the rest of the block. Unless I'm missing something you could drop the else:
```suggestion
        end
        signer_strs_found.delete(s['key'])
        signers_found.add(s)
        weight += s['weight']
```